macos/FluentUI/Button.swift

```diff
@@ -0,0 +1,207 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import AppKit
+
+/// Indicates what style our button is drawn as
+@objc public enum ButtonStyle: Int, CaseIterable {
+    case primaryFilled	// Solid fill color
+    case primaryOutline	// Clear fill color, solid outline
+    case borderless		// clear fill color, clear outline
+}
+
+// MARK: - Button
+
+/// A fluent styled button, with hover effects and a corner radius.
+@IBDesignable
+@objc(MSFButton)
+open class Button: NSButton {
+
+	@objc public init(title: String, style: ButtonStyle = .primaryFilled) {
+		super.init(frame: .zero)
+		self.title = title
+		self.style = style
+        initialize()
+    }
+	
+	public override init(frame frameRect: NSRect) {
+        super.init(frame: frameRect)
+        initialize()
+	}
+
+	@available(*, unavailable)
+	required public init?(coder decoder: NSCoder) {
+		preconditionFailure()
+	}
+	
+	open func initialize() {
+		// Do common initialization work
+		isBordered = false
+		wantsLayer = true
+
+		if #available(macOS 10.14, *) {
+			primaryColor = .controlAccentColor
+		} else {
+			primaryColor = .systemBlue
+		}
+		update()
```

Comment:
Can we come up with a more specific name? As code evolves we tend to want to split these up for clarity and perf gains so starting with a more explicit title can do that by design

Reply:
I meant this to be the most generic form of update.. `updateColorsAndStyle()`? or `updateButtonColorsAndStyle`

Reply:
I guess that fits in to the general goal of swiftUI of making UI declarative by pushing all updatable properties together. This seems fair in general, but we should just verify we're not calling this multiple times on loading

Reply:
I rewrote this to remove the need for update() -> Logic is now done in `didSet` for `title`, and usages of `update()` have been replaced with `needsDisplay = true`